Clarify the use of JSON in requests and responses (#1185)
matrix-org/matrix-spec: content/identity-service-api.md

The mandatory baseline for identity server communication in Matrix is
exchanging JSON objects over HTTP APIs. HTTPS is required for
communication.

All `POST` and `PUT` endpoints, with the exception of [`POST

> /_matrix/identity/v2/account/logout`](#post_matrixidentityv2accountlogout),

Author: there's always an exception.

Reviewer: Why is `/account/logout` an exception to this?

Author: I suspect it's because nobody thought about it before, but: it just is, according to both the openapi definition, and the `sydent` implementation.

Author: (and https://spec.matrix.org/v1.3/identity-service-api/#authentication, which says "no ... request body")

Reviewer: It might be nice to put a "for historical reasons" somewhere in that sentence (though I'm not sure where it would fit best) to indicate why it's like that, and that it isn't because that endpoint is actually special (as opposed to the other exceptions (media upload/download) which have actual reasons for not being JSON).  But if there's no nice way to fit that in, it's not the end of the world.

Author: ```suggestion
All `POST` and `PUT` endpoints, with the exception (for historical reasons) of [`POST
/_matrix/identity/v2/account/logout`](#post_matrixidentityv2accountlogout),
```

require the client to supply a request body containing a (potentially empty)

> JSON object. Clients are *not* required to supply a `Content-Type` header.

Author: ```suggestion
JSON object. Clients should supply a `Content-Type` header of `application/json`
for all requests with JSON bodies, but this is not required.
```

> Similarly, all endpoints require the server to return a JSON object.  Servers
> should include a `Content-Type` header of `application/json` for all JSON
> responses.

Author: ```suggestion
Similarly, all endpoints require the server to return a JSON object.  Servers
must include a `Content-Type` header of `application/json` for all JSON
responses.
```
